### PR TITLE
Fix test failures

### LINKED ISSUE #95214: [Flaky] "pod not found" error in kubelet logs when trying the killing a pod in preemption and eviction scenarios in upstream tests. It randomly failed on docker, process isolation containerd, and hyperv isolation containerd runtime
**What happened**:
Different preemption and eviction scenarios started failing on different platform and runtime classes on windows.
hyperv containerd on 2004: https://testgrid.k8s.io/sig-windows-containerd#aks-engine-azure-windows-master-2004-containerd-hyperv-serial-slow
process containerd on 2004: https://testgrid.k8s.io/sig-windows-containerd#aks-engine-azure-windows-master-2004-containerd-serial-slow
process isolated containerd on server 2019: https://testgrid.k8s.io/sig-windows-containerd#aks-engine-azure-windows-1-19-containerd-serial-slow
hypverv containerd on 2019: https://testgrid.k8s.io/sig-windows-containerd#aks-engine-azure-windows-master-containerd-hyperv-serial-slow
docker: https://testgrid.k8s.io/sig-windows-releases#aks-engine-azure-windows-master-staging-serial-slow on 9/23

Since these set of tests started failing around 9/20 on both all forms of Windows Containers represented and so platform changes at the time. Please investigate from kubelet side.

Preemption scenarios:
Tests created 2 pods use 2/3 of node resources. and one of the pod is low pri
Tests create 3rd pod with higher pri and expect to preempt the low pri pod within 1 mins.
but the time takes to kill the low pri pod is about 57 seconds or even one mins so test timeout. 
We can see "pod not found" errors in kubelet logs when trying the killing the pod:
E0924 21:24:01.500471    4852 remote_runtime.go:329] ContainerStatus "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606" from runtime service failed: rpc error: code = NotFound desc = an error occurred when try to find container "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606": not found
W0924 21:24:01.500471    4852 pod_container_deletor.go:52] [pod_container_deletor] DeleteContainer returned error for (id={containerd 50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606}): failed to get container status "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606": rpc error: code = NotFound desc = an error occurred when try to find container "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606": not found

Related kubelet traces for the a life cycle if the pod:
I0924 21:23:50.009737    4852 reflector.go:210] Starting reflector *v1.Secret (0s) from object-"sched-preemption-5816"/"default-token-drbbw"
I0924 21:23:50.111226    4852 reconciler.go:224] operationExecutor.VerifyControllerAttachedVolume started for volume "default-token-drbbw" (UniqueName: "kubernetes.io/secret/14de82fe-6736-441f-975e-7e0ef611b7c1-default-token-drbbw") pod "pod0-sched-preemption-low-priority" (UID: "14de82fe-6736-441f-975e-7e0ef611b7c1") 
I0924 21:23:50.224427    4852 reconciler.go:269] operationExecutor.MountVolume started for volume "default-token-drbbw" (UniqueName: "kubernetes.io/secret/14de82fe-6736-441f-975e-7e0ef611b7c1-default-token-drbbw") pod "pod0-sched-preemption-low-priority" (UID: "14de82fe-6736-441f-975e-7e0ef611b7c1") 
I0924 21:23:50.229443    4852 operation_generator.go:662] MountVolume.SetUp succeeded for volume "default-token-drbbw" (UniqueName: "kubernetes.io/secret/14de82fe-6736-441f-975e-7e0ef611b7c1-default-token-drbbw") pod "pod0-sched-preemption-low-priority" (UID: "14de82fe-6736-441f-975e-7e0ef611b7c1") 
I0924 21:23:50.324694    4852 kuberuntime_manager.go:425] No sandbox for pod "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)" can be found. Need to start a new one
I0924 21:23:50.324694    4852 kuberuntime_sandbox.go:63] Running pod pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1) with RuntimeHandler "runhcs-wcow-hypervisor-19041"
I0924 21:23:54.166901    4852 provider.go:102] Refreshing cache for provider: *credentialprovider.defaultDockerConfigProvider
I0924 21:23:54.166901    4852 provider.go:102] Refreshing cache for provider: *azure.acrProvider
I0924 21:23:54.447703    4852 kubelet.go:1870] SyncLoop (PLEG): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)", event: &pleg.PodLifecycleEvent{ID:"14de82fe-6736-441f-975e-7e0ef611b7c1", Type:"ContainerStarted", Data:"367686b5aec116d449995f98193f3652c9d9884d9b54335e0270dc4dd5ccbd0e"}
I0924 21:23:56.358601    4852 fake_cpu_manager.go:49] [fake cpumanager] AddContainer (pod: pod0-sched-preemption-low-priority, container: pod0-sched-preemption-low-priority, container id: 50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606)
I0924 21:23:56.358601    4852 fake_topology_manager.go:43] [fake topologymanager] AddContainer  pod: &Pod{ObjectMeta:{pod0-sched-preemption-low-priority  sched-preemption-5816  14de82fe-6736-441f-975e-7e0ef611b7c1 4377 0 2020-09-24 21:23:41 +0000 GMT <nil> <nil> map[] map[kubernetes.io/config.seen:2020-09-24T21:23:50.008762600Z kubernetes.io/config.source:api kubernetes.io/psp:privileged] [] []  [{e2e.test Update v1 2020-09-24 21:23:41 +0000 GMT FieldsV1 {"f:spec":{"f:affinity":{".":{},"f:nodeAffinity":{".":{},"f:requiredDuringSchedulingIgnoredDuringExecution":{".":{},"f:nodeSelectorTerms":{}}}},"f:containers":{"k:{\"name\":\"pod0-sched-preemption-low-priority\"}":{".":{},"f:image":{},"f:imagePullPolicy":{},"f:name":{},"f:resources":{".":{},"f:limits":{".":{},"f:scheduling.k8s.io/foo":{}},"f:requests":{".":{},"f:scheduling.k8s.io/foo":{}}},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:enableServiceLinks":{},"f:priorityClassName":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}} {kube-scheduler Update v1 2020-09-24 21:23:41 +0000 GMT FieldsV1 {"f:status":{"f:conditions":{".":{},"k:{\"type\":\"PodScheduled\"}":{".":{},"f:lastProbeTime":{},"f:lastTransitionTime":{},"f:message":{},"f:reason":{},"f:status":{},"f:type":{}}}}}}]},Spec:PodSpec{Volumes:[]Volume{Volume{Name:default-token-drbbw,VolumeSource:VolumeSource{HostPath:nil,EmptyDir:nil,GCEPersistentDisk:nil,AWSElasticBlockStore:nil,GitRepo:nil,Secret:&SecretVolumeSource{SecretName:default-token-drbbw,Items:[]KeyToPath{},DefaultMode:*420,Optional:nil,},NFS:nil,ISCSI:nil,Glusterfs:nil,PersistentVolumeClaim:nil,RBD:nil,FlexVolume:nil,Cinder:nil,CephFS:nil,Flocker:nil,DownwardAPI:nil,FC:nil,AzureFile:nil,ConfigMap:nil,VsphereVolume:nil,Quobyte:nil,AzureDisk:nil,PhotonPersistentDisk:nil,PortworxVolume:nil,ScaleIO:nil,Projected:nil,StorageOS:nil,CSI:nil,Ephemeral:nil,},},},Containers:[]Container{Container{Name:pod0-sched-preemption-low-priority,Image:e2eteam/pause:3.2-windows-amd64-2004,Command:[],Args:[],WorkingDir:,Ports:[]ContainerPort{},Env:[]EnvVar{},Resources:ResourceRequirements{Limits:ResourceList{scheduling.k8s.io/foo: {{2 0} {<nil>} 2 DecimalSI},},Requests:ResourceList{scheduling.k8s.io/foo: {{2 0} {<nil>} 2 DecimalSI},},},VolumeMounts:[]VolumeMount{VolumeMount{Name:default-token-drbbw,ReadOnly:true,MountPath:/var/run/secrets/kubernetes.io/serviceaccount,SubPath:,MountPropagation:nil,SubPathExpr:,},},LivenessProbe:nil,ReadinessProbe:nil,Lifecycle:nil,TerminationMessagePath:/dev/termination-log,ImagePullPolicy:IfNotPresent,SecurityContext:nil,Stdin:false,StdinOnce:false,TTY:false,EnvFrom:[]EnvFromSource{},TerminationMessagePolicy:File,VolumeDevices:[]VolumeDevice{},StartupProbe:nil,},},RestartPolicy:Always,TerminationGracePeriodSeconds:*1,ActiveDeadlineSeconds:nil,DNSPolicy:ClusterFirst,NodeSelector:map[string]string{kubernetes.io/arch: amd64,kubernetes.io/os: windows,node.kubernetes.io/windows-build: 10.0.19041,},ServiceAccountName:default,DeprecatedServiceAccount:default,NodeName:5450k8s000,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:&PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,SeccompProfile:nil,},ImagePullSecrets:[]LocalObjectReference{},Hostname:,Subdomain:,Affinity:&Affinity{NodeAffinity:&NodeAffinity{RequiredDuringSchedulingIgnoredDuringExecution:&NodeSelector{NodeSelectorTerms:[]NodeSelectorTerm{NodeSelectorTerm{MatchExpressions:[]NodeSelectorRequirement{},MatchFields:[]NodeSelectorRequirement{NodeSelectorRequirement{Key:metadata.name,Operator:In,Values:[5450k8s000],},},},},},PreferredDuringSchedulingIgnoredDuringExecution:[]PreferredSchedulingTerm{},},PodAffinity:nil,PodAntiAffinity:nil,},SchedulerName:default-scheduler,InitContainers:[]Container{},AutomountServiceAccountToken:nil,Tolerations:[]Toleration{Toleration{Key:node.kubernetes.io/not-ready,Operator:Exists,Value:,Effect:NoExecute,TolerationSeconds:*300,},Toleration{Key:node.kubernetes.io/unreachable,Operator:Exists,Value:,Effect:NoExecute,TolerationSeconds:*300,},Toleration{Key:scheduling.k8s.io/foo,Operator:Exists,Value:,Effect:NoSchedule,TolerationSeconds:nil,},Toleration{Key:os,Operator:Equal,Value:windows,Effect:NoSchedule,TolerationSeconds:nil,},},HostAliases:[]HostAlias{},PriorityClassName:sched-preemption-low-priority,Priority:*1,DNSConfig:nil,ShareProcessNamespace:nil,ReadinessGates:[]PodReadinessGate{},RuntimeClassName:*windows-2004,EnableServiceLinks:*true,PreemptionPolicy:*PreemptLowerPriority,Overhead:ResourceList{},TopologySpreadConstraints:[]TopologySpreadConstraint{},EphemeralContainers:[]EphemeralContainer{},SetHostnameAsFQDN:nil,},Status:PodStatus{Phase:Pending,Conditions:[]PodCondition{PodCondition{Type:PodScheduled,Status:True,LastProbeTime:0001-01-01 00:00:00 +0000 UTC,LastTransitionTime:2020-09-24 21:23:50 +0000 GMT,Reason:,Message:,},},Message:,Reason:,HostIP:,PodIP:,StartTime:<nil>,ContainerStatuses:[]ContainerStatus{},QOSClass:BestEffort,InitContainerStatuses:[]ContainerStatus{},NominatedNodeName:,PodIPs:[]PodIP{},EphemeralContainerStatuses:[]ContainerStatus{},},} container id:  50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606
I0924 21:23:58.469409    4852 kubelet.go:1870] SyncLoop (PLEG): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)", event: &pleg.PodLifecycleEvent{ID:"14de82fe-6736-441f-975e-7e0ef611b7c1", Type:"ContainerStarted", Data:"50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606"}
I0924 21:24:00.034263    4852 kubelet.go:1848] SyncLoop (DELETE, "api"): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)"
I0924 21:24:00.473125    4852 kuberuntime_container.go:636] Killing container "containerd://50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606" with a 2 second grace period
I0924 21:24:01.476465    4852 kubelet.go:1870] SyncLoop (PLEG): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)", event: &pleg.PodLifecycleEvent{ID:"14de82fe-6736-441f-975e-7e0ef611b7c1", Type:"ContainerDied", Data:"50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606"}
I0924 21:24:01.476465    4852 kubelet.go:1870] SyncLoop (PLEG): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)", event: &pleg.PodLifecycleEvent{ID:"14de82fe-6736-441f-975e-7e0ef611b7c1", Type:"ContainerDied", Data:"367686b5aec116d449995f98193f3652c9d9884d9b54335e0270dc4dd5ccbd0e"}
I0924 21:24:01.476465    4852 fake_topology_manager.go:48] [fake topologymanager] RemoveContainer container id:  50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606
I0924 21:24:01.499490    4852 fake_topology_manager.go:48] [fake topologymanager] RemoveContainer container id:  50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606
E0924 21:24:01.500471    4852 remote_runtime.go:329] ContainerStatus "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606" from runtime service failed: rpc error: code = NotFound desc = an error occurred when try to find container "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606": not found
W0924 21:24:01.500471    4852 pod_container_deletor.go:52] [pod_container_deletor] DeleteContainer returned error for (id={containerd 50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606}): failed to get container status "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606": rpc error: code = NotFound desc = an error occurred when try to find container "50bc950d407de36fa2cda2b90594db3ba0f1673c04dd64e37ba58bc2e0110606": not found
I0924 21:24:02.685804    4852 reconciler.go:196] operationExecutor.UnmountVolume started for volume "default-token-drbbw" (UniqueName: "kubernetes.io/secret/14de82fe-6736-441f-975e-7e0ef611b7c1-default-token-drbbw") pod "14de82fe-6736-441f-975e-7e0ef611b7c1" (UID: "14de82fe-6736-441f-975e-7e0ef611b7c1") 
W0924 21:24:02.685804    4852 empty_dir.go:453] Warning: Failed to clear quota on c:\var\lib\kubelet\pods\14de82fe-6736-441f-975e-7e0ef611b7c1\volumes\kubernetes.io~secret\default-token-drbbw: not implemented
I0924 21:24:02.687817    4852 operation_generator.go:787] UnmountVolume.TearDown succeeded for volume "kubernetes.io/secret/14de82fe-6736-441f-975e-7e0ef611b7c1-default-token-drbbw" (OuterVolumeSpecName: "default-token-drbbw") pod "14de82fe-6736-441f-975e-7e0ef611b7c1" (UID: "14de82fe-6736-441f-975e-7e0ef611b7c1"). InnerVolumeSpecName "default-token-drbbw". PluginName "kubernetes.io/secret", VolumeGidValue ""
I0924 21:24:02.785865    4852 reconciler.go:319] Volume detached for volume "default-token-drbbw" (UniqueName: "kubernetes.io/secret/14de82fe-6736-441f-975e-7e0ef611b7c1-default-token-drbbw") on node "5450k8s000" DevicePath ""
I0924 21:25:02.157075    4852 kubelet.go:1832] SyncLoop (ADD, "api"): "daemon-set-ck9t9_daemonsets-5408(885abb78-0c4b-4297-a8c2-9cfc985c87e6)"
I0924 21:25:02.157075    4852 reflector.go:210] Starting reflector *v1.Secret (0s) from object-"daemonsets-5408"/"default-token-cc46g"
I0924 21:25:02.302323    4852 reconciler.go:224] operationExecutor.VerifyControllerAttachedVolume started for volume "default-token-cc46g" (UniqueName: "kubernetes.io/secret/885abb78-0c4b-4297-a8c2-9cfc985c87e6-default-token-cc46g") pod "daemon-set-ck9t9" (UID: "885abb78-0c4b-4297-a8c2-9cfc985c87e6") 
I0924 21:25:02.415423    4852 reconciler.go:269] operationExecutor.MountVolume started for volume "default-token-cc46g" (UniqueName: "kubernetes.io/secret/885abb78-0c4b-4297-a8c2-9cfc985c87e6-default-token-cc46g") pod "daemon-set-ck9t9" (UID: "885abb78-0c4b-4297-a8c2-9cfc985c87e6") 
I0924 21:25:02.420424    4852 operation_generator.go:662] MountVolume.SetUp succeeded for volume "default-token-cc46g" (UniqueName: "kubernetes.io/secret/885abb78-0c4b-4297-a8c2-9cfc985c87e6-default-token-cc46g") pod "daemon-set-ck9t9" (UID: "885abb78-0c4b-4297-a8c2-9cfc985c87e6") 
I0924 21:25:02.458478    4852 kuberuntime_manager.go:425] No sandbox for pod "daemon-set-ck9t9_daemonsets-5408(885abb78-0c4b-4297-a8c2-9cfc985c87e6)" can be found. Need to start a new one
I0924 21:25:02.458478    4852 kuberuntime_sandbox.go:63] Running pod daemon-set-ck9t9_daemonsets-5408(885abb78-0c4b-4297-a8c2-9cfc985c87e6) with RuntimeHandler "runhcs-wcow-hypervisor-19041"
I0924 21:25:03.873451    4852 kubelet.go:1848] SyncLoop (DELETE, "api"): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)"
I0924 21:25:03.892448    4852 kubelet.go:1842] SyncLoop (REMOVE, "api"): "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)"
I0924 21:25:03.892448    4852 reflector.go:216] Stopping reflector *v1.Secret (0s) from object-"sched-preemption-5816"/"default-token-drbbw"
I0924 21:25:03.893446    4852 kubelet.go:2039] Failed to delete pod "pod0-sched-preemption-low-priority_sched-preemption-5816(14de82fe-6736-441f-975e-7e0ef611b7c1)", err: pod not found

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.19.0 and v1,20
- Cloud provider or hardware configuration: Standard D2 v3 and Standard D4 v3
- OS (e.g: `cat /etc/os-release`): Windows
- Kernel (e.g. `uname -a`): Linux k8s-master-27464512-0 5.3.0-1032-azure #33~18.04.1-Ubuntu SMP Fri Jun 26 15:01:15 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: n/a
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What type of PR is this?**
/kind cleanup
/kind flake

**What this PR does / why we need it**:
We're not waiting for the pods to be properly deleted in the test suite causing tests to fail especially in case of [Windows](https://testgrid.k8s.io/sig-windows-releases#aks-engine-azure-windows-master-staging-serial-slow).

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```
/sig scheduling
/sig windows

### CODE REVIEW

File: test/e2e/scheduling/preemption.go [hunk 1]
```diff
@@ -287,9 +302,24 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
-			(err == nil && preemptedPod.DeletionTimestamp != nil)
+		var podPreempted bool
+		// Wait till pod gets deleted
+		err := wait.PollImmediate(framework.Poll, framework.PodDeleteTimeout, func() (bool, error) {
+			preemptedPod, err := cs.CoreV1().Pods(pods[0].Namespace).Get(context.TODO(), pods[0].Name, metav1.GetOptions{})
+			// Ignore transient errors
+			if err != nil {
+				return false, nil
+			}
+			if preemptedPod.DeletionTimestamp != nil {
```
Comment (damemi) [inline test/e2e/scheduling/preemption.go:313]: same as above

File: test/e2e/scheduling/preemption.go [hunk 2]
```diff
@@ -186,10 +186,25 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
+		// Make sure that the lowest priority pod gets deleted
+		var podPreempted bool
+		// Wait till pod gets deleted
+		err := wait.PollImmediate(framework.Poll, framework.PodDeleteTimeout, func() (bool, error) {
+			preemptedPod, err := cs.CoreV1().Pods(pods[0].Namespace).Get(context.TODO(), pods[0].Name, metav1.GetOptions{})
+			// Ignore transient errors
+			if err != nil {
+				return false, nil
+			}
+			if preemptedPod.DeletionTimestamp != nil {
```
[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/preemption.go:198]: Won't that be covered [here](https://github.com/kubernetes/kubernetes/pull/96807/files#diff-48c3b5f1a5e8c2a2b246eb43a6efc90351ca6f15ebf948e930acd20d327df0f7R195)?

[the remark this answers is not in the document]
Comment (damemi) [reply test/e2e/scheduling/preemption.go:201]: cool, was just checking. you could also probably clean this whole block up to one if/else with that same logic:

```go
if (err != nil && apierrors.IsNotFound(err)) || (err == nil && preemptedPod.DeletionTimestamp != nil) {
  podPreempted = true
  return podPreempted, nil
} else {
  return false, err
}
```

[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/preemption.go:201]: Ya, I guess I initially authored this test and having that big if condition has confused me as well, so separating it seemed cleaner, however I don't have a strong preference, if you think it makes review easier, I'll change it.

File: test/e2e/scheduling/preemption.go [hunk 3]
```diff
@@ -186,10 +186,21 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
 		})
+		// Make sure that the lowest priority pod gets deleted
+		var podPreempted bool
+		// Wait till pod gets deleted
+		err := wait.PollImmediate(framework.Poll, framework.PodDeleteTimeout, func() (bool, error) {
+			preemptedPod, err := cs.CoreV1().Pods(pods[0].Namespace).Get(context.TODO(), pods[0].Name, metav1.GetOptions{})
+			if (err != nil && apierrors.IsNotFound(err)) || (err == nil && preemptedPod.DeletionTimestamp != nil) {
+				podPreempted = true
+				return podPreempted, nil
+			} else {
```
Comment (damemi) [inline test/e2e/scheduling/preemption.go:197]: sorry, based on golint you can drop the else here and just return false,err outside the if

Comment (damemi) [reply test/e2e/scheduling/preemption.go:197]: same on the other one

**Review by damemi — APPROVED** [review]
/lgtm

looks like it needs a milestone tag though, so ping @ahg-g for that if he wants :)

File: test/e2e/scheduling/preemption.go [hunk 4]
```diff
@@ -186,10 +186,20 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
+		// Wait till pod gets deleted
+		err := wait.PollImmediate(framework.Poll, framework.PodDeleteTimeout, func() (bool, error) {
+			preemptedPod, err := cs.CoreV1().Pods(pods[0].Namespace).Get(context.TODO(), pods[0].Name, metav1.GetOptions{})
+			if (err != nil && apierrors.IsNotFound(err)) || (err == nil && preemptedPod.DeletionTimestamp != nil) {
+				podPreempted = true
+				return podPreempted, nil
+			}
+			// Ignore transient errors
+			return false, err
+		})
```
Comment (spiffxp) [inline test/e2e/scheduling/preemption.go:200]: Can you pull this out into  a separate helper function? e.g. `podPreempted, err := waitForPodPreempted(pods[0].Namespace, pods[0].Name)`?

**Review by spiffxp — CHANGES_REQUESTED** [review]
/area conformance
Impacts a conformance test

I'm less clear whether this is required to land for v1.20 or can wait until v1.20.1 or v1.21

File: test/e2e/scheduling/preemption.go (hunk 4 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/preemption.go:200]: Sure.

**Review by ravisantoshgudimetla — COMMENTED** [review]
> I'm less clear whether this is required to land for v1.20 or can wait until v1.20.1 or v1.21

When do we plan to cut v1.20.1. I am working on these tests to make the test grid go green for Windows tests. The sooner the release date for v1.20.1, the better it is.

**Comment by spiffxp** [comment]
/remove-lifecycle stale
If you can rebase to resolve merge conflicts I will approve

**Comment by jsturtevant** [comment]
spoke with @ravisantoshgudimetla and @aravindhp,  this is no longer needed

/close

**Pull request closed without merging**